toolchain/check/subst.h

```diff
@@ -5,6 +5,7 @@
 #ifndef CARBON_TOOLCHAIN_CHECK_SUBST_H_
 #define CARBON_TOOLCHAIN_CHECK_SUBST_H_
 
+#include "common/map.h"
```

Comment:
I may be missing something, but why is this added? Maybe leftover while refactoring? Perhaps ironically to me, things like llvm::function_ref maybe _should_ have an include.

```suggestion
#include "llvm/ADT/STLFunctionalExtras.h"
```

Reply:
Yeah, left over from refactoring. Just removing the include since I'm also going to look at switching from lambdas to virtual functions.